pkg/porter/dependencies.go

```diff
@@ -248,20 +249,34 @@ func (e *dependencyExecutioner) executeDependency(dep *queuedDependency, parentA
 		// For now, assume it's okay to give the dependency the same credentials as the parent
 		CredentialIdentifiers: parentArgs.CredentialIdentifiers,
 	}
+
+	var executeErrs error
 	fmt.Fprintf(e.Out, "Executing dependency %s...\n", dep.Alias)
 	err := e.CNAB.Execute(depArgs)
 	if err != nil {
-		return errors.Wrapf(err, "error executing dependency %s", dep.Alias)
+		executeErrs = multierror.Append(executeErrs, errors.Wrapf(err, "error executing dependency %s", dep.Alias))
+
+		if depArgs.Action == claim.ActionUninstall {
+			if e.parentOpts.Delete && !e.parentOpts.ForceDelete {
+				executeErrs = multierror.Append(executeErrs, fmt.Errorf("not deleting installation %s as uninstall was not successful; use --force-delete to override", depArgs.Installation))
```

Comment:
Can you update this error message to match the other as well that you just updated?